java/src/main/java/com/logicalclocks/hsfs/FeatureView.java

```diff
@@ -183,31 +185,34 @@ public List<List<Object>> previewFeatureVectors(Integer n, boolean external)
 
   @JsonIgnore
   public String getBatchQuery() throws FeatureStoreException, IOException, ParseException {
-    return getBatchQuery(null, null);
+    return getBatchQuery(null, null, null);
   }
 
   @JsonIgnore
-  public String getBatchQuery(String startTime, String endTime)
+  public String getBatchQuery(String startTime, String endTime, Integer trainingDataVersion)
       throws FeatureStoreException, IOException, ParseException {
     return featureViewEngine.getBatchQueryString(
         this,
         startTime != null ? FeatureGroupUtils.getDateFromDateString(startTime) : null,
-        endTime != null ? FeatureGroupUtils.getDateFromDateString(endTime) : null);
+        endTime != null ? FeatureGroupUtils.getDateFromDateString(endTime) : null,
+        trainingDataVersion);
   }
 
   @JsonIgnore
-  public Dataset<Row> getBatchData(String startTime, String endTime)
+  public Dataset<Row> getBatchData(String startTime, String endTime, Integer trainingDataVersion)
       throws FeatureStoreException, IOException, ParseException {
-    return getBatchData(startTime, endTime, Maps.newHashMap());
+    return getBatchData(startTime, endTime, Maps.newHashMap(), trainingDataVersion);
   }
 
-  public Dataset<Row> getBatchData(String startTime, String endTime, Map<String, String> readOptions)
+  public Dataset<Row> getBatchData(String startTime, String endTime, Map<String, String> readOptions,
```

Comment:
One thing about filters for training data is that they should only be applied when creating the training data. 
When i get batch data for scoring, often i don't want the filters to be applied.
If I don't supply the TD_Version here, will I get all the batch data in the time interval?

Reply:
I have the surfing example, where i can walk through the use case with you to explain why i only want the filter on the TD data, not on the batch data for scoring.

Reply:
> If I don't supply the TD_Version here, will I get all the batch data in the time interval?
yes, td version is optional in the current setting, you can pass `null` here.
> explain why i only want the filter on the TD data, not on the batch data for scoring.
To do this, we need to introduce a new flag "ignore_td_filter" because by default, when td contains statistics as well. If you do not provide td version, no transformation will occur. But I think having the extra may not be necessary. In your case, you can create a new td without filter, and get batch data from there.

Reply:
Good point!
We would still want to apply the transformations and use the TD_version statistics. 
So a new flag - ignore_td_filter, would do the job. Is there a limit to 1 filter or can you add many filters? (ignore_td_filters)

We talked at the feature monitoring meeting last week about having a separate way to log the untransformed features - i know it's not directly related, but you might want to add an explicit flag for that - ignore_transformations.

Reply:
Passing null as a parameter doesn't feel right though. Can we make it an optional parameter in Python (In java, it's not as big a deal, IMO).